libraries/AP_HAL_PX4/CAN.cpp

```diff
@@ -948,39 +1017,47 @@ int PX4CANManager::init(const uint32_t bitrate, const PX4CAN::OperatingMode mode
 
 PX4CAN* PX4CANManager::getIface(uint8_t iface_index)
 {
-    if (iface_index < CAN_STM32_NUM_IFACES) {
+    if (iface_index < _ifaces_num) {
         return ifaces[iface_index];
     }
+
+    return nullptr;
+}
+
+PX4CAN* PX4CANManager::getIface_out_to_in(uint8_t iface_index)
+{
+    // Find which internal interface corresponds to required outside physical interface
+    if (iface_index < CAN_STM32_NUM_IFACES) {
```

Comment:
This is translation from physical interface index into internal. Physical index can be anything up to CAN_STM32_NUM_IFACES. _ifaces_num  is number of internals.